frontend/src/pages/CompareV2.tsx

```diff
@@ -292,9 +313,91 @@ function getDropdownItems(filteredRunArtifacts: RunArtifact[]) {
+  filteredRunArtifacts: RunArtifact[],
+  selectedItem: SelectedItem,
+): Artifact | undefined {
+  const filteredRunArtifact = filteredRunArtifacts.find(
+    runArtifact => runArtifact.run.run?.name === selectedItem.itemName,
+  );
+
+  const executionArtifact = filteredRunArtifact?.executionArtifacts.find(executionArtifact => {
+    const executionText: string =
+      getExecutionName(executionArtifact.execution) ||
+      executionArtifact.execution.getId().toString();
+    return executionText === selectedItem.subItemName;
+  });
+
+  const artifact = executionArtifact?.linkedArtifacts.find(linkedArtifact => {
+    const linkedArtifactText: string =
+      getArtifactName(linkedArtifact) || linkedArtifact.artifact.getId().toString();
+    return linkedArtifactText === selectedItem.subItemSecondaryName;
+  })?.artifact;
+
+  return artifact;
+}
+
 function MetricsDropdown(props: MetricsDropdownProps) {
-  const { filteredRunArtifacts, metricsTabText } = props;
-  const [selectedItem, setSelectedItem] = useState<SelectedItem>({ itemName: '', subItemName: '' });
+  const {
+    filteredRunArtifacts,
+    metricsTab,
+    metricsTabText,
+    selectedArtifacts,
+    setSelectedArtifacts,
+  } = props;
+  const firstSelectedArtifact: SelectedArtifact = selectedArtifacts[metricsTab][0];
+  const secondSelectedArtifact: SelectedArtifact = selectedArtifacts[metricsTab][1];
+
+  const [firstSelectedItem, setFirstSelectedItem] = useState<SelectedItem>(
+    firstSelectedArtifact.selectedItem,
+  );
+  const [secondSelectedItem, setSecondSelectedItem] = useState<SelectedItem>(
+    secondSelectedArtifact.selectedItem,
+  );
+
+  useEffect(() => {
+    if (
+      firstSelectedItem.itemName &&
+      firstSelectedItem.subItemName &&
+      firstSelectedItem.subItemSecondaryName &&
+      selectedArtifacts[metricsTab][0].selectedItem !== firstSelectedItem
+    ) {
+      selectedArtifacts[metricsTab][0].selectedItem = firstSelectedItem;
+      selectedArtifacts[metricsTab][0].selectedArtifact = getArtifactFromSelectedItem(
+        filteredRunArtifacts,
+        firstSelectedItem,
+      );
+      setSelectedArtifacts({ ...selectedArtifacts });
+    }
+  }, [
+    firstSelectedItem,
+    filteredRunArtifacts,
+    metricsTab,
+    selectedArtifacts,
+    setSelectedArtifacts,
+  ]);
+
+  useEffect(() => {
+    if (
+      secondSelectedItem.itemName &&
+      secondSelectedItem.subItemName &&
+      secondSelectedItem.subItemSecondaryName &&
+      selectedArtifacts[metricsTab][1].selectedItem !== secondSelectedItem
+    ) {
+      selectedArtifacts[metricsTab][1].selectedItem = secondSelectedItem;
+      selectedArtifacts[metricsTab][1].selectedArtifact = getArtifactFromSelectedItem(
+        filteredRunArtifacts,
+        secondSelectedItem,
+      );
+      setSelectedArtifacts({ ...selectedArtifacts });
+    }
+  }, [
+    secondSelectedItem,
+    filteredRunArtifacts,
+    metricsTab,
+    selectedArtifacts,
+    setSelectedArtifacts,
+  ]);
```

Comment:
I used two separate `useEffect` hooks because I want to trigger the update for each one independently. This approach (and separate state variables as well) is not sustainable if we allow a dynamic table in the future - should I keep this or try to identify a new solution now?

Reply:
I've simplified this setup by moving the shared code within a `useCallback` hook at https://github.com/kubeflow/pipelines/pull/7966/commits/9d7b93328eedcd6149d578a8e87ff38b33bbb7c6; I think this should improve readability and improve this approach.